enterprise/server/auth/auth.go

```diff
@@ -644,12 +644,19 @@ func authContextFromClaims(ctx context.Context, claims *Claims, err error) conte
 	return ctx
 }
 
-func (a *OpenIDAuthenticator) ParseAPIKeyFromString(input string) string {
+func (a *OpenIDAuthenticator) ParseAPIKeyFromString(input string) (string, error) {
 	matches := apiKeyRegex.FindStringSubmatch(input)
-	if matches != nil && len(matches) > 1 {
-		return matches[1]
+	if len(matches) == 0 {
+		// The api key header is not present
+		return "", nil
 	}
-	return ""
+	if len(matches) != 2 {
+		return "", status.FailedPreconditionError("invalid input")
+	}
+	if apiKey := matches[1]; apiKey != "" {
+		return apiKey, nil
+	}
+	return "", status.FailedPreconditionError("missing API Key")
```

Comment:
nit: do you want to return:

status.UnauthenticatedError("failed to parse API key: invalid input")
and 
status.UnauthenticatedError("failed to parse API key: missing API Key")

here instead, and then just return the err directly in your calling code?


(in general I like to return as detailed of an error as possible at the source, and then just pass ideally without modification in most cases)

Reply:
Done.